090-integrate-bridge-with-metrics-reporter.md

```diff
@@ -0,0 +1,121 @@
+
+The kafka_bridge_config_generator.sh script is used to generate the image configuration based on environment variables.
+This script will be also updated to include `bridge.metrics` and related configurations (see the following section).
+
+The Bridge will try to load the Exporter's configuration file from the path specified by the `bridge.metrics.jmx.exporter.config.path` property.
+If the property is not specified or the file is not found, the Bridge will fall back to the hard coded configuration.
+This feature is not strictly required to support the *Reporter*, but will be used by the Cluster Operator.
+
+### Cluster Operator integration
+
+The KafkaBridge's CRD will also support `metricsConfig` with the addition of `strimziMetricsReporter` type.
+This is how the Strimzi Reporter configuration will look like:
+
+```sh
+spec:
+  metricsConfig:
+    type: strimziMetricsReporter
+    values:
+      allowList:
+        - "kafka_log.*"
+        - "kafka_network.*"
+```
+
+Three new environment variables will be introduced to pass the metrics configuration to the Bridge's container:
+
+- `STRIMZI_METRICS`: This will contain the `metricsConfig` types to enable (one of `jmxPrometheusExporter` and `strimziMetricsReporter`).
+- `KAFKA_BRIDGE_METRICS_JMX_CONFIG`: Used with *Exporter* to pass the configuration file path.
+- `KAFKA_BRIDGE_METRICS_SMR_CONFIG`: Used with *Reporter* to pass the plugin configuration.
+
+The *Exporter* configuration file will be stored in a ConfigMap and mounted in the Bridge's container.
+The full configuration file passed to the Bridge's container will be `/opt/strimzi/custom-config/metrics-config.yml`.
+
+Metrics will be exposed through the Bridge's HTTP server, so the *Reporter*'s listener will be disabled.
+Other configurations will be locked down with the exception of the `prometheus.metrics.reporter.allowlist` property.
+
+## Affected/not affected projects
+
+The affected projects are Cluster Operator and Kafka Bridge.
+
+## Compatibility
+
+### Bridge
+
+The `KAFKA_BRIDGE_METRICS_ENABLED` environment variable will be deprecated in 0.32.0 and removed in 0.34.0.
+When set, the user will get a warning suggesting to use the `bridge.metrics` property.
+In case they are both set, `bridge.metrics` will take precedence over `KAFKA_BRIDGE_METRICS_ENABLED`.
+
+### Cluster Operator
+
+The `enableMetrics` property in KafkaBridge CRD will be deprecated in 0.46.0 and removed in 0.48.0.
+When set, the user will get a warning suggesting to use the `metricsConfig` configuration.
+In case they are both set, `metricsConfig` will take precedence over `enableMetrics`.
+
+## Rejected alternatives
+
+Lock down and automate the *Reporter* configurations for the standalone Bridge.
+This was rejected because the Bridge allows user to customize Kafka clients and plugins using the application.properties file, which includes commented examples.
```

Comment:
```suggestion
This was rejected because the Bridge allows users to customize Kafka clients and plugins using the application.properties file, which includes commented examples.
```